src/js/events/MetronomeActions.js

```diff
@@ -22,201 +20,134 @@ import {
 } from "../constants/ActionTypes";
 import AppDispatcher from "./AppDispatcher";
 import Config from "../config/Config";
+import * as MetronomeClient from "./MetronomeClient";
 
 const MetronomeActions = {
   createJob(data) {
-    RequestUtil.json({
-      url: `${Config.metronomeAPI}/v0/scheduled-jobs`,
-      method: "POST",
-      data,
-      success() {
+    MetronomeClient.createJob(data).subscribe({
```

Comment:
Usually subscribe directly without handling the unsubscription is as bad as registering a callback without handling removing it. The reason is that infinite observers can cause a leak.

```js
const stream = myStream.subscribe(...);

// do the work

stream.unsubscribe();
```
I think here it is fine though because when the request completes, it is going to complete the observer (which does the unsubscribe). The problem I am still thinking here is that if the connection never finishes for some weird reason, `request` does not have a timeout. So  I wonder if we wanna add the timeout on http-service, or if we want have some type of timeout here.

https://github.com/dcos-labs/http-service/blob/89c781cfb54107ed45715d079c4b71149ee24b09/src/request.js

I think I need @orlandohohmeier opinion here.

Reply:
I talked with orlando. The connection has a timeout, which makes the request observable have one too. Which means that we do not care to unsubscribe. So this was never a problem.